### ISSUE TITLE
Exclude artifacts when using globbing

### ISSUE THREAD

**Issue by obones** [actions/attest-build-provenance#150]
Hello,

I'm using the `upload-artifact` action to upload only a few selected files as artifacts, by using such a step:

```yaml
      - name: Upload artifact
        uses: actions/upload-artifact@v4
        with:
          name: com.obones.binding.openmeteo-openhab_${{ matrix.addons-ref }}-java_${{ matrix.java }}
          path: |
            openhab-addons/bundles/com.obones.binding.openmeteo/target/com.obones.binding.openmeteo*.jar
            !openhab-addons/bundles/com.obones.binding.openmeteo/target/com.obones.binding.openmeteo*-sources.jar
```

This allows to upload `com.obones.binding.openmeteo-0.1.0.jar` but not `com.obones.binding.openmeteo-0.1.0-sources.jar` as it is of no use in my case.

I tried using the same glob patterns on the `attest-build-provenance` step like this:

```yaml
      - name: Create attestation
        uses: actions/attest-build-provenance@v1
        with:
          subject-path: |
            openhab-addons/bundles/com.obones.binding.openmeteo/target/com.obones.binding.openmeteo*.jar
            !openhab-addons/bundles/com.obones.binding.openmeteo/target/com.obones.binding.openmeteo*-sources.jar
```

But it still creates two attestations as can be seen [here](https://github.com/obones/openhab-binding-openmeteo/actions/runs/9840624432)

Did I miss something?

**Comment by AliSajid** [comment]
I am also running into the same issue. My `dist/` folder has both the files and the GPG signatures. I'd like to exclude the GPG signatures from being attested.

**Comment by Ahsan3202** [comment]
- name: Create attestation
        uses: actions/attest-build-provenance@v1
        with:
          subject-path: |
            openhab-addons/bundles/com.obones.binding.openmeteo/target/com.obones.binding.openmeteo*.jar
            !openhab-addons/bundles/com.obones.binding.openmeteo/target/com.obones.binding.openmeteo*-sources.jar

**Comment by bdehamer** [comment]
Thanks for the bug report. This _should_ work the same way it does for `upload-artifact`, but I think we may have an issue with the way we're doing the globbing. Will work on a fix and cut a new release.

**Comment by bdehamer** [comment]
Fixed in v1.3.3
